wsd/apps/rest/v0/views/post.py

```diff
@@ -179,3 +181,23 @@ def bookmark(self, *args, **kwargs):
     def unbookmark(self, *args, **kwargs):
         self.request.user.unbookmark(self.get_object())
         return Response(status=204)
+
+    @extend_schema(
+        summary="Get Random Post ID",
+        description="Returns a random post id from the database.",
+        responses={200: str},
+    )
+    @action(
+        detail=False,
+        methods=["GET"],
+        url_path="random",
+        serializer_class=fake_serializer("RandomPost", dont_initialize=True),
+        permission_classes=[ReadOnly],
+    )
+    def get_random_post(self, *args, **kwargs):
+        """
+        Returns a random post id from the database.
+        """
+
+        post_id_: UUID = Post.objects.order_by(Func(function="RANDOM")).first().id
+        return Response(str(post_id_))
```

Comment:
I think this should return a complete post, serializer, like what /posts/<some-uuid> would 

Also `.first().id` is potentially an error if there are no posts, so it'd be good to handle that too

Reply:
I currently have no idea to implement this in a decent way ( without state management ) without having to double the Meme component

Reply:
If you just return the full post here, 

```ts
const { data: postId } = await wsd.getRandomPostId()
``` 
this becomes 
```ts
const { data: post } = await wsd.getRandomPostId()
```

and instead of using `postId` you just use `post.id`. 

Or am I missing something obvious?

Reply:
But if you say that it is ok, then i am fine with it too.

Reply:
I think that's okay, I'd prefer a cleaner API for a tiny performance sacrifice

Reply:
Yeah i guess it wont be used "super often" just occasionally.